### PR TITLE
e2e: wait for pods with gomega

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

gomega.Eventually provides better progress reports: instead of filling up the
log with one-line messages that are not enough to to understand the
current state, it integrates with Gingko's progress reporting (SIGUSR1,
--poll-progress-after) and then dumps the same complete failure message as
after a timeout. That makes it possible to understand why progress isn't
getting made without having to wait for the timeout.
    
The code also is generic enough that it becomes more reusable:
- error handling (= retry after "not found", delay) can be used
  for all kind of objects
- multiple checks can be combined with gomega.And

#### Which issue(s) this PR fixes:
Related-to #https://github.com/kubernetes/kubernetes/issues/106575

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/wait.go
Comment (aojea) [inline test/e2e/framework/wait.go:32]: heh, so this is the first Generic in the codebase? 😄

Comment (pohly) [reply test/e2e/framework/wait.go:32]: It's still in the race, but https://github.com/kubernetes/kubernetes/pull/112377 will probably be faster :grin:

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by dims** [comment]
This PR has the label work-in-progress, please revisit to see if you still need this, please close if not

**Comment by pohly** [comment]
I'll get back to it soonish after the other E2E enhancements have landed.

**Comment by pohly** [comment]
There are some more tests left which (incorrectly) use `framework.ExpectError` and wait for pods to be running which are not expected to start. I'll fix this next week.

File: test/e2e/framework/expect.go
Comment (pohly) [inline test/e2e/framework/expect.go:164]: Prow jobs are currently failing, most likely because the default polling interval is very short and thus causes API server overload.

My intention is to get PR #114783 merged, then use the same defaults here as for the global Gomega in that PR, i.e. it will not be necessary to override the polling interval on every usage of this API here.

**Comment by pohly** [comment]
Another example of a (currently) poor failure message:
```
{ failed [FAILED] while waiting for hostPath init pod to succeed: error while waiting for pod provisioning-6092/hostpath-symlink-prep-provisioning-6092 to be Succeeded or Failed: pod "hostpath-symlink-prep-provisioning-6092" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.4 PodIP:10.244.2.86 PodIPs:[{IP:10.244.2.86}] StartTime:2023-01-09 20:21:32 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:init-volume-provisioning-6092 State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-01-09 20:21:34 +0000 UTC,FinishedAt:2023-01-09 20:21:34 +0000 UTC,ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b Started:0xc0046ccddf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}
In [It] at: test/e2e/storage/drivers/in_tree.go:842 @ 01/09/23 20:21:58.367
}
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/114917/pull-kubernetes-e2e-kind/1612536279373713408

**Comment by pohly** [comment]
And another one:

```
Jan  9 20:21:58.367: INFO: Unexpected error: while waiting for hostPath init pod to succeed: 
    <*fmt.wrapError | 0xc0002738c0>: {
        msg: "error while waiting for pod provisioning-6092/hostpath-symlink-prep-provisioning-6092 to be Succeeded or Failed: pod \"hostpath-symlink-prep-provisioning-6092\" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.4 PodIP:10.244.2.86 PodIPs:[{IP:10.244.2.86}] StartTime:2023-01-09 20:21:32 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:init-volume-provisioning-6092 State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-01-09 20:21:34 +0000 UTC,FinishedAt:2023-01-09 20:21:34 +0000 UTC,ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b Started:0xc0046ccddf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}",
        err: <*errors.errorString | 0xc000dcc6d0>{
            s: "pod \"hostpath-symlink-prep-provisioning-6092\" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.4 PodIP:10.244.2.86 PodIPs:[{IP:10.244.2.86}] StartTime:2023-01-09 20:21:32 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:init-volume-provisioning-6092 State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-01-09 20:21:34 +0000 UTC,FinishedAt:2023-01-09 20:21:34 +0000 UTC,ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b Started:0xc0046ccddf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}",
        },
    }
[FAILED] while waiting for hostPath init pod to succeed: error while waiting for pod provisioning-6092/hostpath-symlink-prep-provisioning-6092 to be Succeeded or Failed: pod "hostpath-symlink-prep-provisioning-6092" failed with status: {Phase:Failed Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:} {Type:Ready Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:ContainersReady Status:False LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason:PodFailed Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2023-01-09 20:21:32 +0000 UTC Reason: Message:}] Message: Reason: NominatedNodeName: HostIP:172.18.0.4 PodIP:10.244.2.86 PodIPs:[{IP:10.244.2.86}] StartTime:2023-01-09 20:21:32 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:init-volume-provisioning-6092 State:{Waiting:nil Running:nil Terminated:&ContainerStateTerminated{ExitCode:1,Signal:0,Reason:Error,Message:,StartedAt:2023-01-09 20:21:34 +0000 UTC,FinishedAt:2023-01-09 20:21:34 +0000 UTC,ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b,}} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:false RestartCount:0 Image:registry.k8s.io/e2e-test-images/busybox:1.29-4 ImageID:registry.k8s.io/e2e-test-images/busybox@sha256:2e0f836850e09b8b7cc937681d6194537a09fbd5f6b9e08f4d646a85128e8937 ContainerID:containerd://4ee3befa399e0ea04144aff31f737ee7c5266846ec904df9f0cb8715f4b3a67b Started:0xc0046ccddf}] QOSClass:BestEffort EphemeralContainerStatuses:[]}
```

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/114917/pull-kubernetes-e2e-kind/1612536279373713408 (stderr of the failed test).

**Comment by claudiubelu** [comment]
/remove-sig windows

Add us if you need something specific to Windows.

**Comment by pohly** [comment]
@aojea: this is ready for review. I cleaned up the commit history, so each commit can stand on its own.

**Comment by SergeyKanzhelev** [comment]
@xmcqueen for sig node lgtm

**Review by xmcqueen — CHANGES_REQUESTED** [review]
I left one comment.  Its probably nothing. I think I'd like you to doublecheck that one thing.  Thanks

File: test/e2e/storage/flexvolume.go
Comment (pohly) [reply test/e2e/storage/flexvolume.go:202]: The commit message for this change explains it:

https://github.com/kubernetes/kubernetes/pull/113298/commits/b3366ce8952daae4a69865134e3180825107cebc

> Calling WaitForPodTerminatedInNamespace after testFlexVolume is useless because
> the client pod that it waits for always gets deleted by testVolumeClient:
> 
> https://github.com/kubernetes/kubernetes/blob/0fcc3dbd55732c501d4948840f639e00a2fe33c5/test/e2e/framework/volume/fixtures.go#L541-L546
> 
> Worse, because WaitForPodTerminatedInNamespace treats "not found" as "must keep
> polling", these two tests always kept waiting for 5 minutes:
> 
>     Kubernetes e2e suite: [It] [sig-storage] Flexvolumes should be mountable
>     when non-attachable 	6m4s
> 
> The only reason why these tests passed is that WaitForPodTerminatedInNamespace
> used to return the "not found" API error. That is not guaranteed and about to
> change.

**Review by xmcqueen — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/hold

Let's hold this over the weekend before merging it, in case that others also have an opinion.

@aojea: last chance to comment :grin:

**Comment by aojea** [comment]
> Let's hold this over the weekend before merging it, in case that others also have an opinion.

I don't have time for review this large PR sorry, but I trust you and the reviewers so feel free to unhold.

My advice, the one that I always do :), merge on weekend and let it soak for at least 1 week, and monitor the CI daily too look for possible changes

Thanks

**Comment by pohly** [comment]
Okay, let's merge it then today.

/hold cancel

**Comment by pohly** [comment]
https://github.com/kubernetes/kubernetes/pull/113298/commits/6eea1b2efa095bfae444849d9a8981307f79dfe8 failed to swap pod name and namespace when replacing WaitForPodToDisappear with WaitForPodNotFoundInNamespace.

Making the API safer to use by using the same order everywhere (horribly messed up right now!) or not putting two strings next to each other at all is out of scope. Let's fix the commit and the merge again:
```diff
diff --git a/test/e2e/framework/network/utils.go b/test/e2e/framework/network/utils.go
index c0023eaf889..e4f733a57e6 100644
--- a/test/e2e/framework/network/utils.go
+++ b/test/e2e/framework/network/utils.go
@@ -888,7 +888,7 @@ func (config *NetworkingTestConfig) DeleteNetProxyPod(ctx context.Context) {
 	framework.ExpectNoError(config.getPodClient().Delete(ctx, pod.Name, *metav1.NewDeleteOptions(0)))
 	config.EndpointPods = config.EndpointPods[1:]
 	// wait for pod being deleted.
-	err := e2epod.WaitForPodNotFoundInNamespace(ctx, config.f.ClientSet, config.Namespace, pod.Name, wait.ForeverTestTimeout)
+	err := e2epod.WaitForPodNotFoundInNamespace(ctx, config.f.ClientSet, pod.Name, config.Namespace, wait.ForeverTestTimeout)
 	if err != nil {
 		framework.Failf("Failed to delete %s pod: %v", pod.Name, err)
 	}
diff --git a/test/e2e/framework/pod/pod_client.go b/test/e2e/framework/pod/pod_client.go
index 8c323d5a3e7..946587ae1df 100644
--- a/test/e2e/framework/pod/pod_client.go
+++ b/test/e2e/framework/pod/pod_client.go
@@ -181,7 +181,7 @@ func (c *PodClient) DeleteSync(ctx context.Context, name string, options metav1.
 	if err != nil && !apierrors.IsNotFound(err) {
 		framework.Failf("Failed to delete pod %q: %v", name, err)
 	}
-	framework.ExpectNoError(WaitForPodNotFoundInNamespace(ctx, c.f.ClientSet, namespace, name, timeout), "wait for pod %q to disappear", name)
+	framework.ExpectNoError(WaitForPodNotFoundInNamespace(ctx, c.f.ClientSet, name, namespace, timeout), "wait for pod %q to disappear", name)
 }
 
 // mungeSpec apply test-suite specific transformations to the pod spec.
diff --git a/test/e2e/framework/volume/fixtures.go b/test/e2e/framework/volume/fixtures.go
index 3f33a5ec123..d8b62b5b0dc 100644
--- a/test/e2e/framework/volume/fixtures.go
+++ b/test/e2e/framework/volume/fixtures.go
@@ -463,7 +463,7 @@ func runVolumeTesterPod(ctx context.Context, client clientset.Interface, timeout
 	}
 	if err != nil {
 		e2epod.DeletePodOrFail(ctx, client, clientPod.Namespace, clientPod.Name)
-		_ = e2epod.WaitForPodNotFoundInNamespace(ctx, client, clientPod.Namespace, clientPod.Name, timeouts.PodDelete)
+		_ = e2epod.WaitForPodNotFoundInNamespace(ctx, client, clientPod.Name, clientPod.Namespace, timeouts.PodDelete)
 		return nil, err
 	}
 	return clientPod, nil
@@ -541,7 +541,7 @@ func testVolumeClient(ctx context.Context, f *framework.Framework, config TestCo
 		// testVolumeClient might get used more than once per test, therefore
 		// we have to clean up before returning.
 		e2epod.DeletePodOrFail(ctx, f.ClientSet, clientPod.Namespace, clientPod.Name)
-		framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, clientPod.Namespace, clientPod.Name, timeouts.PodDelete))
+		framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, clientPod.Name, clientPod.Namespace, timeouts.PodDelete))
 	}()
 
 	testVolumeContent(f, clientPod, "", fsGroup, fsType, tests)
@@ -576,7 +576,7 @@ func InjectContent(ctx context.Context, f *framework.Framework, config TestConfi
 		// This pod must get deleted before the function returns becaue the test relies on
 		// the volume not being in use.
 		e2epod.DeletePodOrFail(ctx, f.ClientSet, injectorPod.Namespace, injectorPod.Name)
-		framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, injectorPod.Namespace, injectorPod.Name, timeouts.PodDelete))
+		framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, injectorPod.Name, injectorPod.Namespace, timeouts.PodDelete))
 	}()
 
 	ginkgo.By("Writing text file contents in the container.")
diff --git a/test/e2e/kubectl/kubectl.go b/test/e2e/kubectl/kubectl.go
index a8c212bda5c..f5574e83cfd 100644
--- a/test/e2e/kubectl/kubectl.go
+++ b/test/e2e/kubectl/kubectl.go
@@ -734,7 +734,7 @@ metadata:
 				if !strings.Contains(ee.String(), "timed out") {
 					framework.Failf("Missing expected 'timed out' error, got: %#v", ee)
 				}
-				framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, ns, "failure-3", 2*v1.DefaultTerminationGracePeriodSeconds*time.Second))
+				framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, "failure-3", ns, 2*v1.DefaultTerminationGracePeriodSeconds*time.Second))
 			})
 
 			ginkgo.It("[Slow] running a failing command with --leave-stdin-open", func(ctx context.Context) {
diff --git a/test/e2e/storage/pd.go b/test/e2e/storage/pd.go
index 755af5a08f1..8aa86b58a2c 100644
--- a/test/e2e/storage/pd.go
+++ b/test/e2e/storage/pd.go
@@ -195,7 +195,7 @@ var _ = utils.SIGDescribe("Pod Disks [Feature:StorageProvider]", func() {
 					ginkgo.By("deleting host0Pod") // delete this pod before creating next pod
 					framework.ExpectNoError(podClient.Delete(ctx, host0Pod.Name, podDelOpt), "Failed to delete host0Pod")
 					framework.Logf("deleted host0Pod %q", host0Pod.Name)
-					e2epod.WaitForPodNotFoundInNamespace(ctx, cs, host0Pod.Namespace, host0Pod.Name, f.Timeouts.PodDelete)
+					e2epod.WaitForPodNotFoundInNamespace(ctx, cs, host0Pod.Name, host0Pod.Namespace, f.Timeouts.PodDelete)
 					framework.Logf("deleted host0Pod %q disappeared", host0Pod.Name)
 				}
 
diff --git a/test/e2e/storage/testsuites/provisioning.go b/test/e2e/storage/testsuites/provisioning.go
index 4a5dc6e151b..87395546363 100644
--- a/test/e2e/storage/testsuites/provisioning.go
+++ b/test/e2e/storage/testsuites/provisioning.go
@@ -845,7 +845,7 @@ func (t StorageClassTest) TestBindingWaitForFirstConsumerMultiPVC(ctx context.Co
 	framework.ExpectNoError(err)
 	ginkgo.DeferCleanup(func(ctx context.Context) error {
 		e2epod.DeletePodOrFail(ctx, t.Client, pod.Namespace, pod.Name)
-		return e2epod.WaitForPodNotFoundInNamespace(ctx, t.Client, pod.Namespace, pod.Name, t.Timeouts.PodDelete)
+		return e2epod.WaitForPodNotFoundInNamespace(ctx, t.Client, pod.Name, pod.Namespace, t.Timeouts.PodDelete)
 	})
 	if expectUnschedulable {
 		// Verify that no claims are provisioned.
diff --git a/test/e2e/storage/ubernetes_lite_volumes.go b/test/e2e/storage/ubernetes_lite_volumes.go
index 8919a4898e4..252c1015502 100644
--- a/test/e2e/storage/ubernetes_lite_volumes.go
+++ b/test/e2e/storage/ubernetes_lite_volumes.go
@@ -100,7 +100,7 @@ func PodsUseStaticPVsOrFail(ctx context.Context, f *framework.Framework, podCoun
 			go func(config *staticPVTestConfig) {
 				defer ginkgo.GinkgoRecover()
 				defer wg.Done()
-				err := e2epod.WaitForPodNotFoundInNamespace(ctx, c, ns, config.pod.Name, f.Timeouts.PodDelete)
+				err := e2epod.WaitForPodNotFoundInNamespace(ctx, c, config.pod.Name, ns, f.Timeouts.PodDelete)
 				framework.ExpectNoError(err, "while waiting for pod to disappear")
 				errs := e2epv.PVPVCCleanup(ctx, c, ns, config.pv, config.pvc)
 				framework.ExpectNoError(utilerrors.NewAggregate(errs), "while cleaning up PVs and PVCs")
diff --git a/test/e2e/windows/density.go b/test/e2e/windows/density.go
index 76854b7da12..e0ab84ae10f 100644
--- a/test/e2e/windows/density.go
+++ b/test/e2e/windows/density.go
@@ -276,7 +276,7 @@ func deletePodsSync(ctx context.Context, f *framework.Framework, pods []*v1.Pod)
 			err := e2epod.NewPodClient(f).Delete(ctx, pod.ObjectMeta.Name, *metav1.NewDeleteOptions(30))
 			framework.ExpectNoError(err)
 
-			err = e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, f.Namespace.Name, pod.ObjectMeta.Name, 10*time.Minute)
+			err = e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, pod.ObjectMeta.Name, f.Namespace.Name, 10*time.Minute)
 			framework.ExpectNoError(err)
 		}(pod)
 	}
diff --git a/test/e2e_node/node_problem_detector_linux.go b/test/e2e_node/node_problem_detector_linux.go
index 904804cac03..39d25e9765c 100644
--- a/test/e2e_node/node_problem_detector_linux.go
+++ b/test/e2e_node/node_problem_detector_linux.go
@@ -433,7 +433,7 @@ current-context: local-context
 			ginkgo.By("Delete the node problem detector")
 			framework.ExpectNoError(e2epod.NewPodClient(f).Delete(ctx, name, *metav1.NewDeleteOptions(0)))
 			ginkgo.By("Wait for the node problem detector to disappear")
-			gomega.Expect(e2epod.WaitForPodNotFoundInNamespace(ctx, c, ns, name, pollTimeout)).To(gomega.Succeed())
+			gomega.Expect(e2epod.WaitForPodNotFoundInNamespace(ctx, c, name, ns, pollTimeout)).To(gomega.Succeed())
 			ginkgo.By("Delete the config map")
 			framework.ExpectNoError(c.CoreV1().ConfigMaps(ns).Delete(ctx, configName, metav1.DeleteOptions{}))
 			ginkgo.By("Clean up the events")
diff --git a/test/e2e_node/resource_collector.go b/test/e2e_node/resource_collector.go
index 5472213511a..60178b9d764 100644
--- a/test/e2e_node/resource_collector.go
+++ b/test/e2e_node/resource_collector.go
@@ -383,7 +383,7 @@ func deletePodsSync(ctx context.Context, f *framework.Framework, pods []*v1.Pod)
 				framework.Failf("Unexpected error trying to delete pod %s: %v", pod.Name, err)
 			}
 
-			framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, f.Namespace.Name, pod.ObjectMeta.Name, 10*time.Minute))
+			framework.ExpectNoError(e2epod.WaitForPodNotFoundInNamespace(ctx, f.ClientSet, pod.ObjectMeta.Name, f.Namespace.Name, 10*time.Minute))
 		}()
 	}
 	wg.Wait()
```

**Comment by pohly** [comment]
/reopen

Not sure whether this is possible for a reverted PR...

**Pull request merged**
